package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -163,8 +173,9 @@ def _single_frame(self):
         if self.grouping == 'atoms':
             positions = self._ags[0].positions  # faster for atoms
         else:
-            # COM for res/frag/etc
-            positions = np.array([elem.centroid() for elem in self.group])
+            # Centre of geometry for residues, segments, fragments
+            positions = np.array(
+                [elem.atoms.centroid() for elem in self.group])
```

Comment:
The center of geometry has no physical meaning. I would still stick to the center of mass. You can get it with 
`positions  = group.center_of_mass(compound=self.grouping)`

Reply:
Seems like one of those where the original code thought COM but actually attempted (and failed) to do COG.

I agree with COM here too.

This does need to be clearly signposted in the documentation for this class. Can you see if you could add this somewhere in the docstring, and also add a relevant example of using this for there various groupings too please?

Reply:
I have added an example of this now